### PR TITLE
Font features

### LINKED ISSUE #57: Convert OpenType font features

### PR DESCRIPTION
**Pull request by javitonino**
Fixes #57

### CODE REVIEW

File: tests/converter/test_text.py [hunk 1]
```diff
@@ -152,26 +152,61 @@ def test_no_kerning(self):
...
-        TEXT_PLAIN["letterSpacing"] = {"units": "PERCENT", "value": 50}
-        TEXT_PLAIN["textAutoResize"] = "WIDTH_AND_HEIGHT"
-
-        text = convert(TEXT_PLAIN)
+        text = convert(
+            {
+                **TEXT_PLAIN,
+                "letterSpacing": {"units": "PERCENT", "value": 50},
+                "textAutoResize": "WIDTH_AND_HEIGHT",
+            }
```
Comment (javitonino) [inline tests/converter/test_text.py:178]: The previous code was changing the "constant" directly, which is a bit ugly. This instead creates a copy of it with the changed fields, which should be nicer if we ever want to do things like parallel tests.

Comment (esloho) [reply tests/converter/test_text.py:178]: You're absolutely right! There are a few other cases in other tests, I'll apply the same change to all 👍

File: converter/text.py [hunk 2]
```diff
@@ -324,3 +326,30 @@ def text_transformation(fig_text):
...
+    for f in fig_text.get("toggledOnOTFeatures", []):
+        try:
+            fid, on, off = FEATURE_MAPPINGS[f.lower()]
+            sketch_features.append(
+                OTFeature(CTFeatureSelectorIdentifier=on, CTFeatureTypeIdentifier=fid)
+            )
+        except KeyError:
+            unsupported_features.append(f)
+
+    for f in fig_text.get("toggledOffOTFeatures", []):
```
Comment (esloho) [inline converter/text.py:343]: Sounds like these 2 loops could be refactored into a small function that gets the property to check (`toggledOnOTFeatures` or `toggleOffOTFeatures`) and the the value for `CTFeatureSelectorIdentifier` (`on` or `off`) and return the list of unsupported features.

**Review by esloho — APPROVED** [review]
🌈

File: converter/text.py (hunk 2 above)
Comment (javitonino) [reply converter/text.py:343]: Yeah, I tried and didn't like it. You have to pass two parameters to it, the array of fig_text and the index of the on/off value. Additionally, you have to return two lists (sketch_features and unsupported_features) and append them. It ended up looking worse IMHO.

Comment (esloho) [reply converter/text.py:343]: Oh, right. Forget it then :homerdissapears:

**Pull request merged**